### PR TITLE
Adding first pass at conda build recipe for zfp

### PR DESCRIPTION
**Pull request by kmpaul**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by kmpaul** [comment]
I've added the following people as maintainers of this recipe.  Let me know if your approve by giving a +1:

- @maboualidev
- @lindstro
- @GarrettDMorrison

**Comment by lindstro** [comment]
@kmpaul What GitHub branch are you planning on packaging?  Only the develop branch has complete tests.  The release branches have minimal sanity tests in the form of the testzfp executable.  I would expect end users to want to install one of the official releases and not code in development.  I'm seeing commits that mention ctest, which is why I'm asking.

**Comment by kmpaul** [comment]
@lindstro Yes.  I'm currently packaging up release 0.5.5.  I only want to point each package release to a real release on github.  So, no, I'm not pointing anything to the development branch.

**Comment by kmpaul** [comment]
Does the ctest suite not work with the releases?  Only development?  I suppose I don't need to use the ctest suite at all.  I was just trying to add it to the conda forge recipe to verify that the build was correct.  But I could just test to make sure `zfpy` is importable.

**Comment by lindstro** [comment]
ctest will still work on the release branches, but it runs only a handful of tests.

**Comment by kmpaul** [comment]
Ok.  Then that should be enough to make sure it built correctly.  Unless it is excessively long to run on conda forge test platforms, then I'll keep it in.

**Comment by kmpaul** [comment]
Ok.  It's getting there.  I think I need to figure out how to get access to `testzfp` from within the test phase of the conda forge recipe build.

I'll do that tomorrow.

**Comment by kmpaul** [comment]
I can't get the Azure pipelines to actually start again.  This is frustrating.

**Comment by kmpaul** [comment]
@conda-forge-admin, please restart ci

**Comment by kmpaul** [comment]
Need more debugging of the Windows build. It’s annoying not being a Windows user (or not having a Windows box to test on).

File: recipes/zfp/bld.bat [hunk 1]
```diff
@@ -0,0 +1,17 @@
+setlocal EnableDelayedExpansion
+
+:: Make a build folder and change to it.
+mkdir build
+cd build
+
+:: Configure using the CMakeFiles
+cmake -DBUILD_ZFPY=ON -DZFP_WITH_OPENMP=OFF -DCMAKE_INSTALL_PREFIX=${PREFIX} ..
```
Comment (xhochy) [inline recipes/zfp/bld.bat]: Try using `-GNinja` and add `ninja` to the requirements. Doesn't solve all your problems but at least may solve some problems.

Comment (kmpaul) [reply recipes/zfp/bld.bat]: Will do.  Thanks!

**Comment by kmpaul** [comment]
@all: See my note on my commit https://github.com/conda-forge/staged-recipes/pull/9901/commits/9f504091734c6caa1449fdcece927156ac5c1bb4

**Comment by kmpaul** [comment]
Not sure. On the VS14 build, I think I saw DLL issues. So that might fix this. That’s great. We could revive the Windows build after that is merged.

**Comment by kmpaul** [comment]
This is ready to be merged.

**Comment by kmpaul** [comment]
@lindstro:  Just for edification, this conda-forge recipe is for zfp release v0.5.5.  So, we will need to do another conda-forge release when the https://github.com/LLNL/zfp/issues/63 PR is merged.  I.e., we will need a new release.  At that time, I will enable the Windows builds.  Does that make sense for you?

**Comment by lindstro** [comment]
Yes, that makes sense.

**Comment by kmpaul** [comment]
Cool!  Then we are now at the whim of the conda-forge gods to merge this PR and create the feedstock.  Just a waiting game, now.

**Comment by kmpaul** [comment]
@conda-forge/staged-recipes When you get time, can you merge this, please?

File: recipes/zfp/bld.bat [hunk 2]
```diff
@@ -0,0 +1,17 @@
+setlocal EnableDelayedExpansion
+
+:: Make a build folder and change to it.
+mkdir build
+cd build
+
+:: Configure using the CMakeFiles
+cmake -GNinja -DBUILD_ZFPY=ON -DZFP_WITH_OPENMP=OFF -DCMAKE_INSTALL_PREFIX=${PREFIX} ..
```
Comment (jakirkham) [inline recipes/zfp/bld.bat]:
```suggestion
cmake -GNinja -DBUILD_ZFPY=ON -DZFP_WITH_OPENMP=OFF -DCMAKE_INSTALL_PREFIX:PATH=%PREFIX% ..
```

File: recipes/zfp/build.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -e
+
+mkdir build
+cd build
+cmake -DBUILD_ZFPY=ON -DZFP_WITH_OPENMP=OFF -DCMAKE_INSTALL_PREFIX=${PREFIX} ..
```
Comment (isuruf) [inline recipes/zfp/build.sh]: Can you add `CMAKE_INSTALL_LIBDIR=lib`?

File: recipes/zfp/build.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+make
```
Comment (isuruf) [inline recipes/zfp/build.sh]: `make -j${CPU_COUNT}`

File: recipes/zfp/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    - {{ compiler('cxx') }}
+    - cmake
+    - make
+  host:
+    - python
+    - numpy
+    - cython
+  run:
+    - python
+    - numpy
```
Comment (isuruf) [inline recipes/zfp/meta.yaml]: See https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

File: recipes/zfp/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,58 @@
+{% set name = "zfp" %}
```
Comment (isuruf) [inline recipes/zfp/meta.yaml]: You should change the name to `zfpy` since this is the python wrapper.

File: recipes/zfp/build.sh (hunk 3 above)
Comment (kmpaul) [reply recipes/zfp/build.sh]: Done

File: recipes/zfp/build.sh (hunk 4 above)
Comment (kmpaul) [reply recipes/zfp/build.sh]: Done

File: recipes/zfp/meta.yaml (hunk 5 above)
Comment (kmpaul) [reply recipes/zfp/meta.yaml]: Done

File: recipes/zfp/meta.yaml (hunk 6 above)
Comment (kmpaul) [reply recipes/zfp/meta.yaml]: Okay

**Comment by kmpaul** [comment]
@conda-forge/staged-recipes Ok.  I've implemented the suggestions from @isuruf and @jakirkham.  Let me know if there is anything else I need to do before this can be merged.  If not, go ahead and merge.

Thanks!

**Pull request merged**